Framework/Core/include/Framework/InputRecord.h

```diff
@@ -351,8 +351,7 @@ class InputRecord
       return *reinterpret_cast<T const*>(ref.payload);
 
       // implementation (i)
-    } else if constexpr (std::is_pointer<T>::value &&
-                         (is_messageable<typename std::remove_pointer<T>::type>::value || has_root_dictionary<typename std::remove_pointer<T>::type>::value)) {
+    } else if constexpr (std::is_pointer<T>::value && is_messageable<typename std::remove_pointer<T>::type>::value) {
```

Comment:
Hmm, the only other solution, which comes to my mind and tries to avoid confusion, is to have a separate method like `getCopy`, but that would require much more new code and work, but perhaps it would find also other use-cases.